### PR TITLE
MM-37542: support tooltip react components

### PR DESCRIPTION
**Pull request by lieut-data**
#### Summary
To facilitate changing the tooltip text on playbooks depending on the context, support a React component when registering the tooltip text. This was actually already supported downstream, but wasn't correctly supported in the registration api.

#### Ticket Link
Relates-to: https://mattermost.atlassian.net/browse/MM-37542

#### Release Note
```release-note
Support React component in channel header tooltips registered by plugins.
```

### CODE REVIEW

File: plugins/registry.js [hunk 1]
```diff
@@ -125,7 +125,7 @@ export default class PluginRegistry {
             icon: resolveReactElement(icon),
             action,
             dropdownText: resolveReactElement(dropdownText),
-            tooltipText,
+            tooltipText: resolveReactElement(tooltipText),
```
Comment (crspeller) [inline plugins/registry.js:128]: Can we update the comment to state it takes a react element as well?

Comment (lieut-data) [reply plugins/registry.js:128]: Mais oui!

**Review by crspeller — APPROVED** [review]

**Review by agarciamontoro — APPROVED** [review]
Nice!

**Pull request merged**